pkg/controller/database_snapshot.go

```diff
@@ -0,0 +1,309 @@
+package controller
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/appscode/go/wait"
+	"github.com/appscode/log"
+	tapi "github.com/k8sdb/apimachinery/api"
+	tcs "github.com/k8sdb/apimachinery/client/clientset"
+	"github.com/k8sdb/apimachinery/pkg/eventer"
+	kapi "k8s.io/kubernetes/pkg/api"
+	k8serr "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	kbatch "k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type Snapshotter interface {
+	Validate(*tapi.DatabaseSnapshot) error
+	GetDatabaseRuntimeObject(*tapi.DatabaseSnapshot) (runtime.Object, error)
```

Comment:
We need this to set events. Suppose you are taking snapshot for Elastic. You want to add events in that Elastic. So I need this runtime object.